### DISCUSSION TITLE
NuxtLink in PopoverPanel does not close Popover (Nuxt 3)

### DISCUSSION THREAD

**Discussion by nklsw** [tailwindlabs/headlessui#1239] [Help]
Reproduction: [Stackblitz example](https://stackblitz.com/edit/github-as55bc?file=layouts/default.vue)

I got a popover as menu in a Nuxt 3 application with some [NuxtLinks](https://v3.nuxtjs.org/docs/directory-structure/pages#navigation) as menu items. When the NuxtLink gets clicked, the correct page gets renderd, but the popover stays open. When I use a normal `<a>` instead of `<NuxtLink>` it's working like expected and the popover closes on click of the menu item. Is this a bug? Or is there a legit reason why this behavior is not working with NuxtLink? A Nuxt (3) problem?

**Comment by xieyuheng** [comment]
the same reason as this: 
https://headlessui.com/react/menu#integrating-with-next-js

**Comment by xieyuheng** [comment]
I define my own `<Link>` in `components/Link.vue`:

```
<script setup lang="ts">
// We can not use <NuxtLink> with headlessui:
// - https://headlessui.com/react/menu#integrating-with-next-js

defineProps<{ href?: string; target?: string }>()

const router = useRouter()
</script>

<template>
  <a @click="href && router.push({ path: href })" :target="target">
    <slot />
  </a>
</template>
```

**Reply by adriaandotcom** [reply]
Seems to work fine indeed: https://stackblitz.com/edit/github-as55bc-whonwl?file=app.vue

**Comment by andkolli** [comment]
Even simpler, [also from the docs](https://headlessui.com/vue/popover#closing-popovers-manually):
> Alternatively, Popover and PopoverPanel expose a close() slot prop which you can use to imperatively close the panel, say after running an async action[.]

So you can do:
```vue
<PopoverPanel v-slot="{ close }">
  <div>
    <NuxtLink :to="/" @click="close">
      Home
    </NuxtLink>
  </div>
</PopoverPanel>
```

**Reply by anthonypillot** [reply]
Thank you, it works perfectly.

**Reply by gokhantaskan** [reply]
`NuxtLink` does not emit a click event in my codebase (`"nuxt": "^3.12.4"`). I can't use it :/

**Reply by victor-michaux** [reply]
I had the same issue with the Dropdown component and I fix it like this:
```vue
<MenuItem v-slot="{ close }">
  <NuxtLink v-slot="{ href }" custom to="/some/url">
    <a :href="href" @click="close">
      {{ item.name }}
     </a>
  </NuxtLink>
<MenuItem />
```
